### PR TITLE
proxy: handles ListenServe error in api proxy

### PR DESCRIPTION
**Pull request by andrewxhill**
@sanderpick caught this one in my recent textile pr so thought I'd nip it here too

### CODE REVIEW

File: api/server/server.go [hunk 1]
```diff
@@ -236,8 +236,10 @@ func NewServer(conf Config) (*Server, error) {
 		grpcServer.Serve(listener)
 	}()
 
+	errc := make(chan error)
 	go func() {
-		grpcWebProxy.ListenAndServe()
+		errc <- grpcWebProxy.ListenAndServe()
+		close(errc)
```
Comment (jsign) [inline api/server/server.go]: I wonder if a channel is used for a particular reason?
Compared to: `if err := grpcWebProxy.ListenAndServe(); err != nil ` and log error.

cc @sanderpick

Comment (sanderpick) [reply api/server/server.go]: Hmm, good question. I don't remember. Yeah, don't need the channel.

**Comment by andrewxhill** [comment]
fixed that

File: api/server/server.go [hunk 2]
```diff
@@ -237,7 +237,9 @@ func NewServer(conf Config) (*Server, error) {
 	}()
 
 	go func() {
-		grpcWebProxy.ListenAndServe()
+		if err := grpcWebProxy.ListenAndServe(); err != nil {
+			log.Errorf("error starting proxy: %v", err)
```
Comment (jsign) [inline api/server/server.go]: Nit: there's an edge case which isn't an error.
```
if err == http.ErrServerClosed {
					// Not really an error, but Shutdown() or Close() was called.
} 
```

**Comment by jsign** [comment]
Merge away!

**Pull request merged**